tuf/src/database.rs

```diff
@@ -99,21 +100,60 @@ impl<D: DataInterchange> Database<D> {
         })
     }
 
+    /// Create a new [`Database`] struct from a set of metadata that is assumed to be trusted. that
+    /// are used to verify the signed metadata. The signed root metadata must be signed with at
+    /// least a `root_threshold` of the provided root_keys. It is not necessary for the root
+    /// metadata to contain these keys.
+    pub fn from_metadata_with_trusted_keys<'a, I>(
+        metadata: &RawSignedMetadataSet<D>,
+        root_threshold: u32,
+        root_keys: I,
+    ) -> Result<Self>
+    where
+        I: IntoIterator<Item = &'a PublicKey>,
+    {
+        let mut db = if let Some(root) = metadata.root() {
+            Database::from_root_with_trusted_keys(root, root_threshold, root_keys)?
+        } else {
+            return Err(Error::MissingMetadata(Role::Root));
+        };
+
+        db.update_metadata_after_root(metadata)?;
+
+        Ok(db)
+    }
+
+    /// Create a new [`Database`] struct from a set of metadata that is assumed to be trusted.
+    ///
+    /// **WARNING**: This is trust-on-first-use (TOFU) and offers weaker security guarantees than
```

Comment:
Perhaps add a comment about when a developer _should_ use this function.

Reply:
Added one, please let me know if it makes sense.

Reply:
This is very helpful, thanks.